v3/src/components/case-table/collection-table.tsx

```diff
@@ -192,22 +199,166 @@ export const CollectionTable = observer(function CollectionTable(props: IProps)
+      grid.scrollTop += scrollSpeed
+      const scrolledTop = grid.scrollTop + grid.clientHeight - 1
+      scrolledToRowIdx = Math.floor(scrolledTop / kDefaultRowHeight)
+
+    }
+    if (scrolledToRowIdx != null && selectionStartRowIdx!= null && scrolledToRowIdx >= 0 &&
+          (rows?.length && scrolledToRowIdx < rows?.length)) {
+      const newSelectedRows = []
+      const startIdx = Math.min(selectionStartRowIdx, scrolledToRowIdx)
+      const endIdx = Math.max(selectionStartRowIdx, scrolledToRowIdx)
+      for (let i = startIdx; i <= endIdx; i++) {
+        const rowId = collectionCaseIdFromIndex(i, data, collectionId)
+        if (rowId) {
+          newSelectedRows.push(rowId)
+        }
+      }
+      setOrExtendSelection(newSelectedRows, data)
+    }
+  }, 25)
+}, [collectionId, data, rows?.length, selectionStartRowIdx])
+
+const stopAutoScroll = useCallback(() => {
+  if (scrollInterval.current) {
+    clearInterval(scrollInterval.current)
+    scrollInterval.current = null
+  }
+}, [])
+
+  // Helper function to get the row index from a mouse event
+  const getRowIndexFromEvent = useCallback((event: React.PointerEvent) => {
+    const target = event.target as HTMLElement
+    const closestDataCell = target.closest('.codap-data-cell')
+    const caseId = closestDataCell?.className.split(" ").find(c => c.startsWith("rowId-"))?.split("-")[1]
+    const rowIdx = caseId ? collectionCaseIndexFromId(caseId, data, collectionId) : null
+    return rowIdx
+  }, [collectionId, data])
+
+   const handlePointerDown = (event: React.PointerEvent<HTMLDivElement>) => {
+    initialPointerDownPosition.current = { x: event.clientX, y: event.clientY }
+    const startRowIdx = getRowIndexFromEvent(event)
+    if (startRowIdx != null) {
+      setSelectionStartRowIdx(startRowIdx)
+      setIsSelecting(true)
+      startAutoScroll(event.clientY)
+      mouseY.current = event.clientY
+    }
+  }
+
+  const handlePointerMove = (event: React.PointerEvent<HTMLDivElement>) => {
+    if (isSelecting && selectionStartRowIdx !== null) {
+      const currentRowIdx = getRowIndexFromEvent(event as React.PointerEvent)
+      if (currentRowIdx != null) {
+        const newSelectedRows = []
+        const start = Math.min(selectionStartRowIdx, currentRowIdx)
+        const end = Math.max(selectionStartRowIdx, currentRowIdx)
+        for (let i = start; i <= end; i++) {
+          newSelectedRows.push(i)
+        }
+        const selectedCaseIds = newSelectedRows
+                                  .map(idx => collectionCaseIdFromIndex(idx, data, collectionId))
+                                  .filter((id): id is string => id !== undefined)
+        setOrExtendSelection(selectedCaseIds, data)
+      }
+      mouseY.current = event.clientY
+      const grid = gridRef.current?.element
+      if (grid) {
+        const { top, bottom } = grid.getBoundingClientRect()
+        if (mouseY.current < top + 50 || mouseY.current > bottom - 20) {
```

Comment:
More hard-coded numbers... and the top one is different than the one above! You should definitely use named constants for these, and they should be the same constants as the ones from lines 259 and 263.